src/insexpand.c

```diff
@@ -3378,6 +3378,103 @@ f_complete_check(typval_T *argvars UNUSED, typval_T *rettv)
     RedrawingDisabled = save_RedrawingDisabled;
 }
 
+/*
+ * "complete_startcol()" function
+ */
+    void
+f_complete_startcol(typval_T *argvars, typval_T *rettv)
+{
+    linenr_T    lnum;
+    colnr_T     col;
+    char_u      *line = NULL;
+    char_u      *ise = curbuf->b_p_ise != empty_option ? curbuf->b_p_ise : p_ise;
+    int         bytepos = -1;
+    char_u      *s = NULL;
+    regmatch_T  regmatch;
+    char_u      *before_cursor = NULL;
+    char_u      trigger = NUL;
+
+    if (rettv_list_alloc(rettv) == FAIL)
+	return;
+
+    if (argvars[0].v_type == VAR_UNKNOWN)
+    {
+	lnum = curwin->w_cursor.lnum;
+	col = curwin->w_cursor.col;
+    }
+    else if (argvars[1].v_type == VAR_UNKNOWN)
+	return;
+    else
+    {
+	lnum = (linenr_T)tv_get_number(&argvars[0]);
+	col = (colnr_T)tv_get_number(&argvars[1]) - 1;  // 0-based
+	
+	if (lnum < 1 || lnum > curbuf->b_ml.ml_line_count || col < 0)
+	    return;
+    }
+
+    line = ml_get_buf(curbuf, lnum, FALSE);
+    if ((int)col > (int)STRLEN(line))
+	col = (colnr_T)STRLEN(line);
```

Comment:
`ml_get_buf_len()`

Reply:
thought about it. exceeding the line should throw an error.